cmd/minishift/cmd/start.go

```diff
@@ -564,3 +581,168 @@ func IsOpenShiftRunning(driver drivers.Driver) bool {
 
 	return openshift.IsRunning(dockerCommander)
 }
+
+/*
+
+Starting local OpenShift cluster using 'kvm' hypervisor...
+   Memory: 2GB
+   Virtual CPU: 2
+   Diskspace: 20GB persistent storage
+
+*/
+
+/*
+
+Starting locsl OpenShift cluster using '' hypervisor...
+-- Provision hostname
+-- Setting systemd unit files for Docker
+   Restarting Docker ... OK
+-- Mounting hostfolders
+   Mounting 'Disks': //remotemachine/Disks/ ... OK
+-- Provision OpenShift via
+
+*/
+
+/*
+
+ * does the minishift instance have an IP address?
+ * can a remote IP address be reached? (check connectivity)
+ * can a remote host be resolved? (check dns)
+ * is persistent disk mounted
+ * #966 check if driver plugin is available
+
+ */
+
+/*
+
+startupCheck("Checking for ... ", func() bool, shouldFail)
+
+*/
+
+type preflightCheckRunner func() bool
+type preflightCheckWithDriverRunner func(driver drivers.Driver) bool
+
+func succeedsOrFails(execute preflightCheckRunner, message string, exitOnError bool, exitMessage string) {
+	print(message)
+	if execute() {
+		println("OK")
+	} else {
+		println("FAIL")
+		if exitOnError {
+			atexit.ExitWithMessage(1, exitMessage)
+		}
+	}
+}
+
+func succeedsOrFailsWithDriver(execute preflightCheckWithDriverRunner, driver drivers.Driver, message string) {
+	print(message)
+	if execute(driver) {
+		println("OK")
+	} else {
+		println("FAIL")
+	}
+}
+
+func preflightChecksBeforeStartHost() {
+	// TODO: to be split-up in individual tests that can be enabled, disabled or ignored
+
+	// to extract
+	vmdriver := viper.GetString(startFlags.VmDriver.Name)
+	switch vmdriver {
+	case "xhyve":
+		succeedsOrFails(
+			checkXhyveDriver,
+			"-- Checking if 'xhyve' driver installed ... ",
+			false, "")
+	case "kvm":
+		succeedsOrFails(
+			checkKvmDriver,
+			"-- Checking if 'kvm' driver installed ... ",
+			false, "")
+	case "hyperv":
+		succeedsOrFails(checkHypervDriver,
+			"-- Checking of 'hyperv' driver configured ...",
+			true,
+			"   Hyper-V virtual switch not set")
+	}
+}
+
+func checkXhyveDriver() bool {
+	_, err := exec.LookPath("docker-machine-driver-xhyve")
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func checkKvmDriver() bool {
+	_, err := exec.LookPath("docker-machine-driver-kvm")
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func checkHypervDriver() bool {
+	switchEnv := os.Getenv("HYPERV_VIRTUAL_SWITCH")
+	if switchEnv == "" {
+		return false
+	}
+	return true
+}
+
+func preflightChecksAfterStartHost(driver drivers.Driver) {
+	succeedsOrFailsWithDriver(checkIPConnectivity, driver, "-- Checking host '8.8.8.8' reachable from VM ... ")
```

Comment:
As mentioned in #485, values for this test (and the DNS check) are to be configurable for override by config settings. At the moment, `8.8.8.8` seems to be a good test to perform to see if something is remotely reachable. But yes, when proxy is concerned, this might fail. This test is therefore also a WARNING, and not an ERROR.